src/org.hdfgroup.object/hdf/object/Dataset.java

```diff
@@ -165,6 +170,29 @@
     /** The number of data points in the memory buffer. */
     protected long            nPoints             = 1;
 
+    /**
+     * The array to store flags to indicate if a member of this compound
+     * dataset is selected for read/write.
+     * <p>
+     * If a member is selected, the read/write will perform on the member.
+     * Applications such as HDFView will only display the selected members of
+     * the compound dataset.
+     *
+     * <pre>
+     * For example, if a compound dataset has four members
+     *     String[] memberNames = {"X", "Y", "Z", "TIME"};
+     * and
+     *     boolean[] isMemberSelected = {true, false, false, true};
+     * members "X" and "TIME" are selected for read and write.
+     * </pre>
+     */
+    protected boolean[] isMemberSelected;
+
+    /**
+     * The datatypes of the compound attribute's members.
+     */
+    protected Datatype[] memberTypes = null;
```

Comment:
Mentioned from before, these fields and getSelectedMemberCount/getSelectedMemberTypes don't belong here.

Reply:
I will investigate why I changed this.

Reply:
One problem, which was solved with AttributeDataset, is deciding an object in a list or a reference is an Attribute. Pushing all the attribute functions/properties into a "dataset" class structure loses that "attribute" common class between a scalar and compound attribute (assuming the attributes rely on scalarDS/compoundDS respectively).
Maybe we need an attribute interface?

Reply:
A simple instanceof should be able to determine if something is an Attribute. But everything compound related should go back into CompoundDS so that only compound datasets and attributes inherit these fields and methods and not "scalar" ones.